### PR TITLE
create a raisedHands object in config.js

### PR DESCRIPTION
**Pull request by liumengyuan1997**
Hi reviewer,

This is a GSoC task. I added a raisedHands object in config.js and added its corresponding selectors. Feel free to let me know which part is inappropriate.

Thanks,
Mengyuan

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -232,8 +232,14 @@ var config = {
     // Sets the preferred resolution (height) for local video. Defaults to 720.
     // resolution: 720,
 
-    // Specifies whether the raised hand will hide when someone becomes a dominant speaker or not
-    // disableRemoveRaisedHandOnFocus: false,
```
Comment (damencho) [inline config.js:236]: You should not delete this, but deprecate it. Check the other options that have DEPRECATED.  in the comment.
You should also make sure that the old config works in parallel with the new one. That ensures that we are not breaking experience for people with the old config set.

File: react/features/base/config/functions.any.ts [hunk 2]
```diff
@@ -89,7 +89,47 @@ export function getFeatureFlag(state: IReduxState, featureFlag: string) {
...
+    return state['features/base/config']?.raisedHands?.disableRemoveRaisedHandOnFocus || false;
+}
+
+/**
+ * Selector used to get the disableLowerAllHand.
+ *
+ * @param {Object} state - The global state.
+ * @returns {boolean}
+ */
+export function getDisableLowerAllHand(state: IReduxState) {
```
Comment (saghul) [inline react/features/base/config/functions.any.ts:101]: I think we should use a single option for disabling the lower hand functionality. I'm not sure who would want to allow one but not the other...

File: react/features/base/participants/subscriber.ts [hunk 3]
```diff
@@ -33,7 +37,9 @@ StateListenerRegistry.register(
 StateListenerRegistry.register(
     /* selector */ state => state['features/base/participants'].raisedHandsQueue,
     /* listener */ (raisedHandsQueue, store) => {
-        if (raisedHandsQueue.length && isNextToSpeak(store.getState()) && !hasBeenNotified(store.getState())) {
```
Comment (saghul) [inline react/features/base/participants/subscriber.ts:36]: Why remove the extra check?

File: react/features/base/config/functions.any.ts [hunk 4]
```diff
@@ -89,7 +89,47 @@ export function getFeatureFlag(state: IReduxState, featureFlag: string) {
  * @returns {boolean}
  */
 export function getDisableRemoveRaisedHandOnFocus(state: IReduxState) {
-    return state['features/base/config']?.disableRemoveRaisedHandOnFocus || false;
```
Comment (saghul) [inline react/features/base/config/functions.any.ts:92]: The config reducer should allow reading the old value and setting it in the new format.

**Review by saghul — CHANGES_REQUESTED** [review]

File: config.js [hunk 5]
```diff
@@ -232,9 +232,26 @@ var config = {
...
+
+    //     // Specifies whether there is a notification before hiding the raised hand
+    //     // when someone becomes the dominant speaker.
+    //     disableLowerHandNotification: true,
+
+    //     // Specifies whether there is a notification when you are the next speaker in line.
+    //     disableNextSpeakerNotification: false,
+
+    //     // Specifies whether the raised hand will hide when someone becomes a dominant speaker or not.
+    //     disableRemoveRaisedHandOnFocus: false
```
Comment (saghul) [inline config.js:252]: Nit: add a trailing comma so the syntax remains correct even when new options are added after.

**Review by saghul — DISMISSED** [review]

File: config.js (hunk 5 above)
Comment (saghul) [inline config.js:252]:
```suggestion
    //     disableRemoveRaisedHandOnFocus: false,
```

**Review by saghul — APPROVED** [review]

**Pull request merged**